### ISSUE TITLE
Support for Fetch queries that use entityname in condition

### ISSUE THREAD

**Issue by xairrick** [jordimontana82/fake-xrm-easy#284]
Fetch queries allow for linked entity filters to be placed on the top entity by using the 'entityname' attribute  (basically allowing for more complex and/or filters).

When I tried this with fake-xrm-easy   it threw an error indicating that 'name' wasn't an attribute of 'task'.   This query does work against CRM (test against v8.2).
```
var entities = new List<Entity>();
var opportunity = new Opportunity() { Id = Guid.NewGuid(), Name = "Test" };
entities.Add(opportunity);
var task = new Task() { Id = Guid.NewGuid(), Subject = "Test Task", RegardingObjectId= opportunity.ToEntityReference() };
entities.Add(task);
var context = new XrmFakedContext();
context.ProxyTypesAssembly = Assembly.GetAssembly(typeof(Opportunity));
context.Initialize(entities);
var service = context.GetFakedOrganizationService();

var fetch = @"<fetch top='50' >
    <entity name='task' >
    <filter>
            <condition entityname='opportunity' attribute='name' operator='eq' value='Test' />
    </filter>
    <link-entity name='opportunity' from='opportunityid' to='regardingobjectid' link-type='inner' />
    </entity>
</fetch>";
var res = service.RetrieveMultiple(new Microsoft.Xrm.Sdk.Query.FetchExpression(fetch));
```

**Comment by nickbrandl** [comment]
I've come across this issue while trying to test a fetchxml with an outer join that uses the "entityname" element in the condition, and while it doesn't throw an error, it returns unexpected records.
the query that I'm trying to execute is the following:

`<fetch version="1.0" output-format="xml-platform" mapping="logical" distinct="true">
  <entity name="account">
    <attribute name="name" />
    <attribute name="primarycontactid" />
    <attribute name="telephone1" />
    <attribute name="accountid" />
    <order attribute="name" descending="false" />
    <filter type="and">
      <condition attribute="statecode" operator="eq" value="0" />
    </filter>
    <link-entity name="contact" from="parentcustomerid" to="accountid" link-type="outer" alias="ac" />
    <filter type="and">
      <condition entityname="ac" attribute="parentcustomerid" operator="null" />
    </filter>
  </entity>
</fetch>`
This query should fetch all active accounts without related contacts, and it works fine in Dynamics, but with the faked service, it returns all active accounts with or without related contacts, like a simple left join.
I'm using FakeXrmEasy.9.

**Comment by fallasd** [comment]
I have encountered something which may - or may not - be related. 

When I have the following FetchXml, how is the distinct='true' applied? Does it refer to just the primary-entity record and should it return one record? Or, given 4 linked-entity records which match the link-entity filter, should it return 4 records (i.e. should distinct-ness refer to the complete query)?

FakeXrmEasy returns multiple records which is different from a Dynamics CRM production environment which appears to return just one. I am using FakeXrmEasy.365 (v8.2) and FakeXrmEasy.9 (v9.0)

```
<fetch version='1.0' output-format='xml-platform' mapping='logical' distinct='true'>
  <entity name='primary-entity'>
    <attribute name='primary-entity-id />
    <attribute name='an-attribute'/>
    <filter type='and'>
      <condition attribute='an-attribute' operator='eq' value='something' />
    </filter>
    <link-entity name='linked-entity' from='primary-entity' to='primary-entity-id' alias='le'>
      <filter type='and'>
        <condition attribute='an-attribute' operator='in'>
          <value>1</value>
          <value>2</value>
          <value>3</value>
          <value>4</value>
        </condition>
      </filter>
    </link-entity>
  </entity>
</fetch>
```

**Comment by BetimBeja** [comment]
Hello @fallasd,
i have tried the following query and it effectively returning multiple records for the same contact, I have the contact as primary contact in multiple accounts (for testing purposes) and the distinct-ness refers to the complete query:
```xml
<fetch distinct="true" >
  <entity name="contact" >
    <attribute name="fullname" />
    <filter type="and" >
      <condition attribute="statecode" operator="eq" value="0" />
    </filter>
    <link-entity name="account" from="primarycontactid" to="contactid" alias="Account" >
      <attribute name="name" />
    </link-entity>
  </entity>
</fetch>
```
Best regards,
Betim.

**Comment by fallasd** [comment]
@jordimontana Cool, thanks. In the meantime I have patched my SUT code so that it ignores any records after the first.

**Comment by jordimontana82** [comment]
This has been resolved in #411 

Thanks a mill @BetimBeja
